core/src/bank_forks.rs

```diff
@@ -20,6 +20,24 @@ pub struct BankForks {
     root: u64,
     slots: HashSet<u64>,
     snapshot_path: Option<String>,
+    confidence: HashMap<u64, Confidence>,
+}
+
+#[derive(Debug, Default, PartialEq)]
+pub struct Confidence {
+    fork_stakes: u64,
+    epoch_stakes: u64,
+    lockouts: u64,
+}
+
+impl Confidence {
+    pub fn new(fork_stakes: u64, epoch_stakes: u64, lockouts: u64) -> Self {
+        Self {
+            fork_stakes,
+            epoch_stakes,
+            lockouts,
```

Comment:
I think we need to add a stake weighted lockout.  what the user would want to compute is the amount of slashing or reward loss the "stake" will suffer to unroll this block.

Reply:
Will add